libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresher.kt

```diff
@@ -46,8 +48,28 @@ internal object AnnotationsRefresher {
             newAnnotation,
             startingLegGeometryIndex,
         ) { maxspeed() }
-        // unrecognized properties migrate from new annotation
-        return (newAnnotation?.toBuilder() ?: LegAnnotation.builder())
+        val unrecognizedProperties = oldAnnotation.unrecognizedPropertiesNames
+            .union(newAnnotation?.unrecognizedPropertiesNames ?: emptySet())
+            .associateNonNullValuesWith { propertyName ->
+                if (
+                    oldAnnotation.getUnrecognizedProperty(propertyName)?.isJsonArray == false ||
```

Comment:
Why do we discard the new data if original response was empty? If we have new information we should add it to the existing instance.

Reply:
It's an annotation that's supposed to be refreshed using some quite strict rules. If we used current_route_geometry_index, old values are not returned in the refresh response. The response may be truncated so we have to fill the rest with the old values. We could use some default values but for the other annotations we require the original ones to be filled. Even for this case: if we fill everything that's not returned in the refresh response with 0 or 100, it may break some user algorithms based on these annotations.

Reply:
That makes sense :+1: